db/rdbms/migration/0002_migrate_descriptor_to_extended_descriptor.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package migration
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/facebookincubator/contest/pkg/job"
+	"github.com/facebookincubator/contest/pkg/test"
+	"github.com/facebookincubator/contest/tools/migration/rdbms/migrate"
+
+	"github.com/facebookincubator/contest/cmds/plugins"
+	"github.com/facebookincubator/contest/pkg/pluginregistry"
+
+	"github.com/sirupsen/logrus"
+)
+
+const shardSize = uint64(50)
+
+// DescriptorMigration represents a migration which moves steps description in jobs tables from old to new
+// schema. The migration consists in the following:
+//
+//
+// In v0001, job.Request object was structured as follows:
+//
+// type Request struct {
+//		JobID         types.JobID
+//		JobName       string
+//		Requestor     string
+//		ServerID      string
+//		RequestTime   time.Time
+//		JobDescriptor string
+//		TestDescriptors string
+// }
+//
+// Having TestDescriptors as a field of request objects creates several issues, including:
+// * It's an abstraction leakage, the extended description of the steps might not be part
+//   of the initial request submmitted from outside (i.e. might not be a literal embedded
+//	 in the job descriptor)
+// * The job.Descriptor object does not contain the extended version of the test steps, which
+//   will make it difficult to handle resume, as we need to retrieve multiple objects
+//
+// Schmema v0002 introduces the concept of extended_descriptor, which is defined as follows:
+// type ExtendedDescriptor struct {
+//		JobDescriptor
+//		StepsDescriptors []test.StepsDescriptors
+// }
+//
+// We remove TestDescriptors from Request objects, and we store that information side-by-side with
+// JobDescriptor into an ExtendedDescriptor. We then store this ExtendedDescriptor in the jobs table
+// so that all the test information can be re-fetched simply by reading extended_descriptor field in
+// jobs table.
+type DescriptorMigration struct {
+	log *logrus.Entry
+}
+
+func ms(d time.Duration) float64 {
+	return float64(float64(d.Microseconds()) / 1000.0)
```

Comment:
**nitpicking:**

1. Extra `float64()`
2. It could've been `d.Seconds() / 1000` or `float64(d.Nanoseconds()) / float64(time.Millisecond.Nanoseconds())`. Personally I would prefer `d.Seconds() / 1000`.